### ISSUE TITLE
Support for cts:annotation in cts:query?

### ISSUE THREAD

**Issue by ralfhergert** [marklogic/java-client-api#1648]
CtsQueries allow to be enriched with cts:annotation-elements: https://docs.marklogic.com/guide/search-dev/cts_query#id_28038
Those elements are ignored by MarkLogic when the query is executed. cts:annotations can be used for various purposes (like documentation) and in our case we use it as a marker-element to identify certain parts of the query to be able to modify and manipulate them later after the creation.

Do the QueryBuilders support to add annotation-elements? (I searched, but didn't find it yet.)

**Comment by rjrudin** [comment]
Correct, there's not support for adding annotation elements. What's your use case for them? Like would you eventually serialize the query to a place where a developer / user would see it?

**Comment by ralfhergert** [comment]
Hi @rjrudin, we are starting to use alerts in MarkLogic server. The queries used in the alert-rules are composed of two parts: one is given by our customers and the other is given by our rights and roles model. From time to time the rights and roles part is updated and thus we need to update the queries in all alert-rules. Here we only want to modify the half of the query corresponds to the rights and roles, we absolutely do not want to touch the part of the query given by our customers. - Annotations help us to identify the parts correctly.

**Comment by SoumenduGhosh10** [comment]
Hi @ralfhergert is this issue still important for you? Or the workaround solved your problem?

**Comment by ralfhergert** [comment]
@SoumenduGhosh10 we worked around this issue. So for our particular use-case and our implementation this is currently not an issue. (But still this doesn't change the fact that cts:annoation is an allowed and specified element for which there is no representation in the ML java client's model classes. In other words the model classes are incomplete.)

**Comment by rjdew-progress** [comment]
Support for cts:annotation in a cts:query has been added in the 8.2.0 release.
